UKFTest/NoisyPendulum/UKF.py

```diff
@@ -0,0 +1,198 @@
+import scipy.linalg as la
+import numpy as np
+#from numba import jit
+from numpy.linalg import solve, slogdet
+
+
+
+################################
+
+
+################################
+
+#@jit(nopython=True)
+def f(t, x, g):
+    x0 = x[0]
+    x1 = x[1]
+
+    rhs = np.asarray([x1, -g*np.sin(x0)])
+    return rhs
+
+#@jit(nopython=True)
+def rk4_step(func, x, t, dt, *args):
+    k1 = dt * func(t, x, *args)
+    k2 = dt * func(t + dt / 2, x + k1 / 2, *args)
+    k3 = dt * func(t + dt / 2, x + k2 / 2, *args)
+    k4 = dt * func(t + dt, x + k3, *args)
+
+    return x + (k1 + 2 * k2 + 2 * k3 + k4) / 6
+
+#@jit(nopython=True)
+def propagate_sigma_points(SigmaPoints, params, tcur, dt):
+    g = params 
+
+    PropagatedSigmaPoints = np.empty_like(SigmaPoints)
+    for i, x in enumerate(SigmaPoints):
+        PropagatedSigmaPoints[i] = rk4_step(f, x, tcur, dt, g)
+
+    return PropagatedSigmaPoints
+
+################################
+
+#@jit(nopython=True)
+def Predict(Wc, Wm, PropagatedSigmaPoints, nstates, Q):
+    Xp = np.zeros((nstates, 1))
+    for i in range(len(Wm)):
+        Xp += Wm[i] * PropagatedSigmaPoints[i].reshape(nstates, 1)
+
+        
+    SigmaPointDiff = np.zeros((len(PropagatedSigmaPoints), nstates))
```

Comment:
@OlearyJoe Unless I am missing something, this SigmaPointDiff is never actually used, right? It is recalculated in the update stage?